suite-native/discovery/src/discoveryMiddleware.ts

```diff
@@ -69,6 +71,12 @@ export const prepareDiscoveryMiddleware = createMiddlewareWithExtraDeps(
             );
         }
 
+        // ensure that BTC is enabled when device with BTC-only firmware is connected
+        // (it could have been disabled via some other device with universal firmware)
+        if (deviceActions.selectDevice.match(action) && selectHasBitcoinOnlyFirmware(getState())) {
```

Comment:
I'm not sure about the `selectDevice` action. I'm not saying it's wrong choice, just that I want to double check that. Also think about future feature to switch to btc-only FW.

Reply:
So what action do you suggest instead? 🤔

Reply:
As I said, I just wanted to double check that. Now I believe that `selectDevice` is a good choice, since `device-connect` has some edge cases like you switch to view-only universal wallet, disable BTC, go back to bitcoin-only device... 

But you should not use `getState()` here, that could contain the other device than the one being selected. I think you should replace `selectHasBitcoinOnlyFirmware` with `hasBitcoinOnlyFirmware` and pass the payload of the action to it (it contains the device).

Reply:
And we should probably add here one more condition for `device.connected` - we don't want to enable BTC it the btc-only device is not connected but in view-only - because we can't do the discovery in that case.

Reply:
It’s funny how something that looks so simple can be so complex. 😄

Reply:
Fixup pushed, seems to work fine. 👍